merkle/sparse_merkle_tree_test.go

```diff
@@ -443,12 +443,14 @@ func testSparseTreeFetches(ctx context.Context, t *testing.T, vec sparseTestVect
 		readMutex.Lock()
 		defer readMutex.Unlock()
 
-		state, ok := reads[ids[0].String()]
-		reads[ids[0].String()] = "met"
-
-		// Allow re-reads of the same nodes too (state=="met") - this is due to
-		// cache warming:
-		return ok && (state == "unmet" || state == "met")
+		for _, id := range ids {
+			strID := id.String()
+			if reads[strID] != "unmet" {
+				return false
+			}
+			reads[strID] = "met"
```

Comment:
will this line ever be reached?

Reply:
Yes, it must be, otherwise the test will fail.

Reply:
The "unmet" line above will never happen, for the same reason - the test will fail. But it was there for ease of debugging in case it does fail.